cgroup2/utils.go

```diff
@@ -379,41 +389,55 @@ func systemdUnitFromPath(path string) string {
 func readHugeTlbStats(path string) []*stats.HugeTlbStat {
 	usage := []*stats.HugeTlbStat{}
 	keyUsage := make(map[string]*stats.HugeTlbStat)
+
 	f, err := os.Open(path)
 	if err != nil {
 		return usage
 	}
-	files, err := f.Readdir(-1)
+
+	files, err := f.Readdirnames(-1)
 	f.Close()
 	if err != nil {
 		return usage
 	}
 
+	buf := make([]byte, 32)
 	for _, file := range files {
-		if strings.Contains(file.Name(), "hugetlb") &&
-			(strings.HasSuffix(file.Name(), "max") || strings.HasSuffix(file.Name(), "current")) {
-			var hugeTlb *stats.HugeTlbStat
-			var ok bool
-			fileName := strings.Split(file.Name(), ".")
+		if strings.Contains(file, "hugetlb") &&
+			(strings.HasSuffix(file, "max") || strings.HasSuffix(file, "current")) {
+			var (
+				hugeTlb *stats.HugeTlbStat
+				ok      bool
+			)
+			fileName := strings.Split(file, ".")
 			pageSize := fileName[1]
 			if hugeTlb, ok = keyUsage[pageSize]; !ok {
 				hugeTlb = &stats.HugeTlbStat{}
 			}
 			hugeTlb.Pagesize = pageSize
-			out, err := os.ReadFile(filepath.Join(path, file.Name()))
+
+			f, err := os.Open(filepath.Join(path, file))
 			if err != nil {
 				continue
 			}
+			defer f.Close()
```

Comment:
nit: Can/Should we instead try to close the file pointer in the function after reading rather than defering the call?

Reply:
You mean manual f.Close calls if Read fails/if read succeeds?

Reply:
@manugupt1 is right, because otherwise we may have quite a few files opened before we exit from this function.

Yet better -- switch to `getStatFileContentUint64`

Reply:
^ That seems best